unleash-yggdrasil/src/strategy_upgrade.rs

```diff
@@ -207,39 +209,50 @@ fn upgrade_remote_address(strategy: &Strategy) -> String {
                 .join(", ");
             format!("remote_address in [{ips}]")
         }
-        None => "".into(),
+        None => "false".into(),
     }
 }
 
 fn upgrade_session_id_rollout_strategy(strategy: &Strategy) -> String {
-    let percentage = strategy.get_param("percentage");
+    let percentage = get_rollout_percent(strategy);
+
     let group_id = strategy.get_param("groupId");
     match (percentage, group_id) {
         (Some(percentage), Some(group_id)) => {
             format!("{percentage}% sticky on session_id with group_id of \"{group_id}\"")
         }
-        _ => "".into(),
+        _ => "false".into(),
     }
 }
 
 fn upgrade_user_id_rollout_strategy(strategy: &Strategy) -> String {
-    let percentage = strategy.get_param("percentage");
+    let percentage = get_rollout_percent(strategy);
+
     let group_id = strategy.get_param("groupId");
     match (percentage, group_id) {
         (Some(percentage), Some(group_id)) => {
             format!("{percentage}% sticky on user_id with group_id of \"{group_id}\"")
         }
-        _ => "".into(),
+        _ => "false".into(),
     }
 }
 
 fn upgrade_random(strategy: &Strategy) -> String {
-    match strategy.get_param("percentage") {
+    let percentage = get_rollout_percent(strategy);
+
+    match percentage {
         Some(percent) => format!("random < {percent}"),
-        None => "".into(),
+        None => "false".into(),
     }
 }
 
+fn get_rollout_percent(strategy: &Strategy) -> Option<usize> {
```

Comment:
Utility method

Reply:
Should we make this method more generic and make it accept the param name as well? That way you can use for the "rollout" above as well. Perhaps it could be generic enough to also accept the type.

Reply:
Yuh fair, I've added a param name to this so they now share code but I'm very hesitant to bind these together too tightly with an abstraction, I think this is one of those cases where the code is identical but the abstraction is a coincidence